dandischema/models.py

```diff
@@ -1863,25 +1847,16 @@ class Publishable(DandiBaseModel):
     )
 
 
-_doi_field_kwargs: dict[str, Any] = {
-    "title": "DOI",
-    "json_schema_extra": {"readOnly": True, "nskey": DANDI_NSKEY},
-}
-if DANDI_DOI_PATTERN is not None:
-    _doi_field_kwargs["pattern"] = DANDI_DOI_PATTERN
-else:
-    _doi_field_kwargs["default"] = ""
-    # restricting the value to empty string to indicate that there is no DOI
-    _doi_field_kwargs["pattern"] = r"^$"
-
-
 class PublishedDandiset(Dandiset, Publishable):
     id: str = Field(
         description="Uniform resource identifier.",
-        pattern=DANDI_PUBID_PATTERN,
         json_schema_extra={"readOnly": True},
     )
-    doi: str = Field(**_doi_field_kwargs)
+    doi: str = Field(
+        title="DOI",
+        json_schema_extra={"readOnly": True, "nskey": DANDI_NSKEY},
+        default="",
```

Comment:
We require `doi` value when DANDI DOI pattern is not available. Defaulting to `"""` will fail the `check_id` method below, but the feedback the user getting will be different.

Reply:
Ah I see, that's an oversight on my part.